src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheck.java

```diff
@@ -261,6 +296,10 @@ public void visitToken(DetailAST ast) {
         if (isInterfaceOrAnnotationMember(ast)) {
             processInterfaceOrAnnotation(ast);
         }
+
+        if (Integer.parseInt(jdkVersion) >= JDK_17) {
```

Comment:
https://maven.apache.org/plugins/maven-compiler-plugin/examples/set-compiler-source-and-target.html

How we are going to handle 1.x version ?

What if we split by `.` and take as int first digit.
In case 1.x it will be 1, but we don't care until it becomes 17 :).
We are fully backwards compatible until 17, so anything less than 17 is whatever.

Reply:
Yes, we should split 1.x on `.` and take the x. "x" in this case will be any number between 0 and  8. Starting from java 9 it will be 9, 10, 11, .. , 22 only. 

Right now we only support the single-number versions. So people on Java 1.6 should use 6. But we also have the property as a string to keep us flexible if we want to support mapping from 1.x to x. @nrmancuso suggested adding support for this mapping in another issue. Do we need to support it in this PR?

Reply:
My point it that until 17 we don't need versions at all , so any integer part of not number that is less than 17 is the same version, same behavior.
We don't need to have 21 as value in code as it also meaning

Reply:
Yeah, right now we only care if this number is less than 17.

It might be a good idea to abstract the internal representation of this property from the API of the check. We could create a LanguageLevel enum and use logic like this:
 - anything that contains a `.` or up to 16 is LEVEL_LT_17
 - 17 and above can be LEVEL_17

If we separate the language level of the check from the user-facing jdkversion property, we can make "buckets" like this for this check.

Once we add some new logic here, we can just add a new language level enum constant. Each enum can have a `intValue` field.
 -

Reply:
> My point it that until 17 we don't need versions at all , so any integer part of not number that is less than 17 is the same version, same behavior.
 
 Yes exactly whatever version number less than 17 will not affect the check behavior. It will be affected only by 17 and 21 (21 is another issue- unnamed variables)
 
 > We don't need to have 21 as value in code as it also meaningless for us, it is just above 17 - all that we need to care.
 
Yes, for now it is meaningless, but we will need 21 in https://github.com/checkstyle/checkstyle/issues/15159 `if (jdkVersion > 21)` violate final on unnamed

Reply:
Something like this:
```
    enum JavaLanguageLevel {

        LEVEL_LT_17(16),
        LEVEL_17(17),
        ;

        private final int value;

        JavaLanguageLevel(int value) {
            this.value = value;
        }

        public int getIntValue() {
            return value;
        }

        public JavaLanguageLevel getLevel(String jdkVersion) {
            final JavaLanguageLevel level;
            if (jdkVersion.startsWith("1.")
                    || Integer.parseInt(jdkVersion) < 17) {
                level = LEVEL_LT_17;
            } else {
                level = LEVEL_17;
            }
            return level;
        }
    }
```

Reply:
ok looks like we in some agreement, let put this in code to discuss details of implementation.

Reply:
> It might be a good idea to abstract the internal representation of this property from the API of the check. We could create a LanguageLevel enum and use logic like this:

anything that contains a . or up to 16 is LEVEL_LT_17
17 and above can be LEVEL_17


Very good, done